packages/guides/src/Intersphinx/JsonWriter.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Guides\Intersphinx;
+
+use function file_put_contents;
+use function json_encode;
+
+use const JSON_PRETTY_PRINT;
+use const JSON_THROW_ON_ERROR;
+
+class JsonWriter
+{
+    /** @param array<mixed> $json */
+    public function saveJsonToFile(array $json, string $filename): void
+    {
+        $jsonString = json_encode($json, JSON_PRETTY_PRINT, JSON_THROW_ON_ERROR);
+        file_put_contents($filename, $jsonString);
```

Comment:
It would be nice if you could use a `\League\Flysystem\FilesystemInterface` here to write to disk. In phpDocumentor, we are abstracting the filesystem completely because we want to provide options to write to s3 in the future. 

https://flysystem.thephpleague.com/v1/docs/usage/filesystem-api/

If you turn this into a new `\phpDocumentor\Guides\Renderer` implementation it is covered automatically for you.

Reply:
Actually the json writer is not really needed as of yet, so I would remove it and turn to the renderers later on